lib/bothan/metrics.rb

```diff
@@ -114,6 +114,16 @@ def self.registered(app)
         get_metric_range(params)
       end
 
+      app.delete '/metrics/:metric' do
+        protected!
+        Metric.where(name: params[:metric]).each do |metric|
```

Comment:
Best way to do this is use https://apidock.com/rails/ActiveRecord/Relation/destroy_all, which will do it (a) in a single line and (b) do all the callbacks and associations.

Reply:
cheers, thanks for the pointer on this I sensed that my way was a bit of a kludge